xarray/test/test_backends.py

```diff
@@ -149,6 +149,27 @@ def assert_loads(vars=None):
             actual = ds.load()
         self.assertDatasetAllClose(expected, actual)
 
+    def test_dataset_compute(self):
+        expected = create_test_data()
+
+        with self.roundtrip(expected) as actual:
+            # Test Dataset.compute()
+            for v in actual.variables.values():
+                self.assertFalse(v._in_memory)
+
+            computed = actual.compute()
+
+            # indexes are going to be loaded in memory anyway, even
+            # with compute(). This is by design.
```

Comment:
sorry, that was a comment from when I was trying to have IndexVariable do the caching and Variable do not. But then I realised that in some cases the data of a dask-backed array is actually an IndexVariable. Removing the comment.